doc/user/content/manage/troubleshooting.md

```diff
@@ -183,6 +183,32 @@ ORDER BY elapsed_ns DESC
 From the results of this query we can see that most of the elapsed time of the dataflow is caused by the time it takes to maintain the updates in one of the arrangements of the dataflow of the materialized view.
 
 ## Why is Materialize unresponsive or slow?
+Slow responses can be the result of a number of factors, which we can help you understand and address.
+
+If you haven't already, you should consider an [index](https://materialize.com/docs/sql/create-index/) for your query, and check out the [optimizations](https://materialize.com/docs/transform-data/optimization/) on how to optimize query performance in Materialize.
+
+You can also consider changing your isolation level, depending on the [consistency guarantees](https://materialize.com/docs/get-started/isolation-level) that you need.
```

Comment:
This section is mixing recommendations on what to do if Materialize is unresponsive or slow, and root causes. I can push a suggestion to reorganize the content as a follow-up (wouldn't want to block your PR on that).

Reply:
That would be awesome! Hoping to use this as a springboard for other improvements.